bigflow/build/dev.py

```diff
@@ -0,0 +1,98 @@
+"""Build-related developer tools and helpers.
+
+Intented to be used by "CLI" and in "development environement".
+"""
+
+import sys
+import logging
+import tempfile
+import pickle
+import typing
+import os
+import os.path
+
+from typing import Optional, List, Union
+from pathlib import Path
+
+import bigflow.commons as bfc
+import bigflow.build.reflect
+
+
+logger = logging.getLogger(__name__)
+
+# Used by 'dist.py' to dump `bigflow.setup(...)` arguments.
+DUMP_PARAMS_SETUPPY_CMDARG = "__bigflow_dump_params"
+
+
+def read_setuppy_args(path_to_setup: Union[Path, str, None] = None) -> dict:
+    """Loads `setup.py`, returns all parameters of `bigflow.build.setup()` function.
+
+    This function doesn't unpack 'embeeded sdist' archive when package is installed via pip.
+    You could use `bigflow.build.materialize_setuppy` for such purposes, altough it is not recommended"""
+
+    path_to_setup = path_to_setup or find_setuppy()
+    logger.info("Read project options from %s", path_to_setup)
+    with tempfile.NamedTemporaryFile("r+b") as f:
+        bfc.run_process(["python", path_to_setup, DUMP_PARAMS_SETUPPY_CMDARG, f.name], cwd=str(path_to_setup.parent))
+        params = pickle.load(f)
+
+    legacy_project_name = _read_project_name_from_setup_legacy(path_to_setup.parent)
+    if legacy_project_name and params.get('name') != legacy_project_name:
+        logging.error(
+            "Project name mismatch: setup.PROJECT_NAME == %r, "
+            "but setup(name=%r). It is recommended to remove 'PROJECT_NAME' variable from 'project_setup.py'",
+            legacy_project_name, params.get('name'))
+
+    return params
+
+
+def _read_project_name_from_setup_legacy(direcotry: Path) -> Optional[str]:
+    # TODO: Please, remove me!
```

Comment:
?

Reply:
Agree, this is not a good comment. Fixed.